pkg/operation/operation.go

```diff
@@ -599,3 +600,35 @@ func (o *Operation) ComputeIngressHost(prefix string) string {
 	shortID := strings.Replace(o.Shoot.Info.Status.TechnicalID, shoot.TechnicalIDPrefix, "", 1)
 	return fmt.Sprintf("%s-%s.%s", prefix, shortID, o.Seed.IngressDomain())
 }
+
+// ToAdvertisedAddresses returns list of advertised addresses on a Shoot cluster.
+func (o *Operation) ToAdvertisedAddresses() []gardencorev1beta1.ShootAdvertisedAddress {
+	addresses := []gardencorev1beta1.ShootAdvertisedAddress{}
```

Comment:
What about

```suggestion
	var addresses []gardencorev1beta1.ShootAdvertisedAddress
```

to return `nil` when there is no address instead of an empty list?

Reply:
It doesn't matter for the API at the end.

Reply:
True, but I think in most other places we prefer returning `nil` in such cases. If it doesn't make a difference can you adapt?